spring-test/src/main/java/org/springframework/test/context/observation/ObservationThreadLocalTestListener.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2002-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.test.context.observation;
+
+import io.micrometer.observation.ObservationRegistry;
+import io.micrometer.observation.contextpropagation.ObservationThreadLocalAccessor;
+
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.Conventions;
+import org.springframework.test.context.TestContext;
+import org.springframework.test.context.TestExecutionListener;
+
+/**
+ * {@code ObservationThreadLocalTestListener} is an extension of the {@link TestExecutionListener}
+ * SPI that updates the {@link ObservationThreadLocalAccessor} with the {@link ObservationRegistry}
+ * taken from the {@link ApplicationContext} present in the {@link TestContext}.
+ *
+ * This implementation is not thread-safe.
+ *
+ * @author Marcin Grzejszczak
+ * @since 6.0
+ */
+public class ObservationThreadLocalTestListener implements TestExecutionListener {
```

Comment:
```suggestion
public class ObservationThreadLocalTestExecutionListener extends AbstractTestExecutionListener {
```

How do you envision this listener being registered?

Will users have to register it manually via `@TestExecutionListeners`? (I'd typically recommend against that)

Or should this listener be automatically registered as a default (like every other listener in `spring-test`)?

If the latter, this listener needs to be declared in `/spring-test/src/main/resources/META-INF/spring.factories`.

In any case, this listener should extend `AbstractTestExecutionListener` and provide a reasonable "order" value. See other concrete implementations in `spring-test` to determine which order to use (or ask for assistance if needed).